core/src/main/scala/org/apache/spark/SparkConf.scala

```diff
@@ -168,6 +168,15 @@ class SparkConf(loadDefaults: Boolean) extends Cloneable with Logging with Seria
   }
 
   /** Set multiple parameters together */
+  def setAll(settings: Iterable[(String, String)]): SparkConf = {
+    settings.foreach { case (k, v) => set(k, v) }
+    this
+  }
+
+  /**
+   * Set multiple parameters together
+   */
+  @deprecated("Use setAll(Iterable) instead", "3.0.0")
```

Comment:
Hm, is it even worth keeping and deprecating this and removing it later? It'll be source compatible already and binary compatibility isn't guaranteed; one would have to go out of one's way to call the deprecated method at all, with a cast or with some odd custom collection

Reply:
One reason I put this forth is that the `TraversableOnce` problem is harder; its replacement `IterableOnce` doesn't exist in Scala 2.12. The same overload strategy doesn't work out quite right for the places it's used (generic types and function signatures make the overload ambiguous). I am considering proposing just changing the type there for 3.0.

Reply:
I can see setAll() is being used though. but maybe it won't work even after a rebuild (bin compat)?

Reply:
Calls to `setAll` will now bind to the more specific `Iterable` arg version in all cases in Spark, and I'm not aware of any Scala classes that aren't `Iterable` to begin with. One consequence I suppose is that passing an `Iterator` will use the deprecated method. But it'll still be there. I don't anticipate we'll remove it until we want to support Scala 2.13 and that won't be in 3.0. For now, no method is removed so should be no problem.